### ISSUE TITLE
Susie with sparse LD matrices

### ISSUE THREAD

**Issue by nlapier2** [stephenslab/susieR#225]
This is perhaps an atypical usage of susie-rss, but it would be nice if it were able to natively handle sparse LD matrices. 

For example,

```library(Matrix)
zscores = 1:5
R = diag(5)
N = 10000
library(susieR)
res = susie_rss(zscores, R, n=N)
```

runs just fine, but then

```
Rs = Matrix(R, sparse=T)
res = susie_rss(zscores, Rs, n=N)
```

yields 

`Error: Not compatible with requested type: [type=S4; target=double].`

I ran into this issue when attempting to run susie on an LD matrix with a block diagonal structure (because it spanned multiple loci) that I was trying to exploit for faster computation and lower memory usage.

Incidentally, I seem to have been able to use sparse matrices in the [cTWAS](https://github.com/xinhe-lab/ctwas/tree/main) version of susie(-rss) -- not sure why.

**Comment by pcarbo** [comment]
@nlapier2 It appears that `susie_rss` does not currently support "R" being a sparse matrix. 

As a temporary workaround, what about running `susie_rss` separately on each of the blocks?

**Comment by jhchung** [comment]
A separate, but related question when using sparse LD matricies. 

If there are a lot of null entries (e.g getting ld from a precomputed LD reference with minimum R2 cutoffs), i usually set these to 0.  How does the credible set purity calculation take this into account? If it takes the upper triangle, will that mean the minimum purity will be 0?

**Comment by pcarbo** [comment]
@jhchung Correct; if SNPs i, j are in the same CS and R[i,j] = 0, then the minimum correlation of the CS will be zero.
